api/repositories/process_repository_test.go

```diff
@@ -6,249 +6,197 @@ import (
-					Expect(process.HealthCheck.Data.TimeoutSeconds).To(Equal(cfProcess1.Spec.HealthCheck.Data.TimeoutSeconds))
-					Expect(process.HealthCheck.Data.HTTPEndpoint).To(Equal(cfProcess1.Spec.HealthCheck.Data.HTTPEndpoint))
-				})
+			It("returns an untyped error", func() {
+				Expect(getErr).To(HaveOccurred())
+				Expect(getErr).To(MatchError(ContainSubstring("get-process: privileged client list failed")))
 			})
 		})
 
 		When("duplicate Processes exist across namespaces with the same GUIDs", func() {
 			var (
-				cfApp1 *workloadsv1alpha1.CFApp
-				cfApp2 *workloadsv1alpha1.CFApp
-
-				cfProcess1 *workloadsv1alpha1.CFProcess
+				namespace2 *corev1.Namespace
+				app2GUID   string
 				cfProcess2 *workloadsv1alpha1.CFProcess
 			)
 
 			BeforeEach(func() {
-				cfApp1 = initializeAppCR("test-app1", app1GUID, namespace1.Name)
-				Expect(k8sClient.Create(context.Background(), cfApp1)).To(Succeed())
-
-				cfApp2 = initializeAppCR("test-app2", app2GUID, namespace2.Name)
-				Expect(k8sClient.Create(context.Background(), cfApp2)).To(Succeed())
+				app2GUID = prefixedGUID("app2")
 
-				cfProcess1 = initializeProcessCR(process1GUID, namespace1.Name, app1GUID)
-				Expect(k8sClient.Create(context.Background(), cfProcess1)).To(Succeed())
+				namespace2 = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: prefixedGUID("namespace2")}}
+				Expect(k8sClient.Create(context.Background(), namespace2)).To(Succeed())
 
 				cfProcess2 = initializeProcessCR(process1GUID, namespace2.Name, app2GUID)
 				Expect(k8sClient.Create(context.Background(), cfProcess2)).To(Succeed())
 			})
 
-			//AfterEach(func() {
-			//	k8sClient.Delete(context.Background(), cfApp1)
-			//	k8sClient.Delete(context.Background(), cfApp2)
-			//	k8sClient.Delete(context.Background(), cfProcess1)
-			//	k8sClient.Delete(context.Background(), cfProcess2)
-			//})
+			AfterEach(func() {
+				Expect(k8sClient.Delete(context.Background(), namespace2)).To(Succeed())
+			})
 
-			It("returns an error", func() {
-				_, err := processRepo.GetProcess(testCtx, authInfo, process1GUID)
-				Expect(err).To(HaveOccurred())
-				Expect(err).To(MatchError("duplicate processes exist"))
+			It("returns an untyped error", func() {
+				Expect(getErr).To(HaveOccurred())
+				Expect(getErr).To(MatchError("duplicate processes exist"))
 			})
 		})
 
-		When("no Processes exist", func() {
-			It("returns an error", func() {
-				_, err := processRepo.GetProcess(testCtx, authInfo, "i don't exist")
-				Expect(err).To(HaveOccurred())
-				Expect(err).To(MatchError(NotFoundError{ResourceType: "Process"}))
+		When("no matching processes exist", func() {
+			BeforeEach(func() {
+				getProcessGUID = "i don't exist"
+			})
+
+			It("returns a not found error", func() {
+				Expect(getErr).To(HaveOccurred())
+				Expect(getErr).To(MatchError(repositories.NotFoundError{ResourceType: "Process"}))
 			})
 		})
+
+		It("returns a not found error when the user has no permission to see the process", func() {
```

Comment:
This is the new code. The rest is a refactoring to make it possible to write this simply.